### PR TITLE
Don't report duplicate Net-NTLM hashes

### PR DESCRIPTION
**Pull request by zeroSteiner**
Because Net-NTLM (v1 and v2) hashes both include a challenge, the private data Metasploit stores is unique. When a credential object is stored, Metasploit deduplicates it based on the value but because they're unique Metasploit ends up storing Net-NTLM hashes as new entries. This means we're logging data to the database and console that's largely unnecessary. Users only need one full hash with the challenge and response in order to crack the corresponding plaintext offline.

These changes address this problem by looking up the value to see if there is already one in the database from the source IP, user identity (realm/domain + username) and hash type to ensure that only "unique" values are kept. This means when users go to export the values from the database to run a cracking attack, they don't need to duplicate them themselves. It also removes the extra lines from the logged output, replacing it with a verbose-only message stating that a duplicate was received.

## Verification

List the steps needed to make sure this thing works

- [ ] Start `msfconsole`
- [ ] Delete all of the creds out of the current workspace or use a new workspace so there are no pre-existing entries
- [ ] `use auxiliary/server/capture/smb` and run it
- [ ] Authenticate to the service using Windows or RubySMB's `example/authenticate.rb` script
- [ ] See exactly 1 reported entry for a unique combination of host, user identity and hash type
- [ ] Run `creds` and see no duplicate entries

### CODE REVIEW

File: modules/auxiliary/server/capture/smb.rb [hunk 1]
```diff
@@ -17,9 +17,10 @@ def initialize
       'Description' => %q{
         This module provides a SMB service that can be used to capture the challenge-response
         password NTLMv1 & NTLMv2 hashes used with SMB1, SMB2, or SMB3 client systems.
-        Responses sent by this service have by default a random 8 byte challenge string
-        of format `\x11\x22\x33\x44\x55\x66\x77\x88`, allowing for easy cracking using
-        Cain & Abel (NTLMv1) or John the ripper (with jumbo patch).
+        Responses sent by this service by default use a random 8 byte challenge string.
+        A specific value (such as `1122334455667788`) can be set using the CHALLENGE option,
+        allowing for easy cracking using Cain & Abel (NTLMv1) or John the Ripper
+        (with jumbo patch).
```
Comment (zeroSteiner) [inline modules/auxiliary/server/capture/smb.rb:23]: This isn't actually related to the problem statement in the PR description, but I realized while looking through the module that it's no longer accurate. When we updated the module we switched the default to be random which is more difficult to identify as malicious. These changes reflect the current state of things.

**Review by cdelafuente-r7 — APPROVED** [review]

File: lib/msf/core/exploit/remote/smb/server/hash_capture.rb [hunk 2]
```diff
@@ -103,9 +95,31 @@ def report_ntlm_type3(address:, ntlm_type1:, ntlm_type2:, ntlm_type3:)
       # found_host.os_name = credential_options[:client_os_version]
       # found_host.save!
 
+      search_options = {
+        ptype: credential_options[:jtr_format],
```
Comment (jmartin-r7) [inline lib/msf/core/exploit/remote/smb/server/hash_capture.rb:99]: What is the reasoning to have `pytpe` and `jtr_format` both reference the same value?

File: lib/msf/core/exploit/remote/smb/server/hash_capture.rb [hunk 3]
```diff
@@ -103,9 +97,31 @@ def report_ntlm_type3(address:, ntlm_type1:, ntlm_type2:, ntlm_type3:)
...
```
Comment (jheysel-r7) [inline lib/msf/core/exploit/remote/smb/server/hash_capture.rb:101]:
```suggestion
```

**Comment by jheysel-r7** [comment]
Changes look good and testing was as expected I also removed the duplicate `credential_options[:jtr_format]` from the `search_options` hash as I found it unnecessary. Thanks Spencer! 

```
msf6 auxiliary(server/capture/smb) > run
[*] Auxiliary module running as background job 0.
msf6 auxiliary(server/capture/smb) >
[*] Server is running. Listening on 0.0.0.0:445
[*] Server started.
[+] Received SMB connection on Auth Capture Server!

msf6 auxiliary(server/capture/smb) > creds
Credentials
===========

host            origin          service        public         private                                                                                              realm  private_type        JtR Format
----            ------          -------        ------         -------                                                                                              -----  ------------        ----------
172.16.199.234  172.16.199.234  445/tcp (smb)  Administrator  Administrator:::f1d8b68c249387cc:9af2e893e9d7cdb90a6efa26d43e78b2:0101000000000000009aa (TRUNCATED)         Nonreplayable hash  netntlmv2

msf6 auxiliary(server/capture/smb) >
[+] Received SMB connection on Auth Capture Server!

msf6 auxiliary(server/capture/smb) > creds
Credentials
===========

host            origin          service        public         private                                                                                              realm  private_type        JtR Format
----            ------          -------        ------         -------                                                                                              -----  ------------        ----------
172.16.199.234  172.16.199.234  445/tcp (smb)  Administrator  Administrator:::f1d8b68c249387cc:9af2e893e9d7cdb90a6efa26d43e78b2:0101000000000000009aa (TRUNCATED)         Nonreplayable hash  netntlmv2
```

**Comment by jheysel-r7** [comment]
# Release Notes
The `auxiliary/server/capture/smb` module no longer stores duplicate Net-NTLM hashes in the database.

**Pull request merged**